src/utilities.jl

```diff
@@ -0,0 +1,89 @@
+using MCMCChains, StatsFuns
+export pointwise_loglikes, compute_loo
+
+"""
+    pointwise_loglikes(chain::Chains, data, ll_fun)
+
+Computes the pointwise log likelihoods where [d,s,c] corresponds to log likelihood of 
+evaluated at datapoint d, sample s, for chain c. 
+
+- `chain`: a chain object from MCMCChains
+- `data`: a vector of data used to estimate parameters of the model 
+- `ll_fun`: a function that computes the log likelihood of a single data point: f(θ1, ..., θn, data), where θi is the ith parameter
+"""
+function pointwise_loglikes(chain::Chains, data, ll_fun)
+    samples = Array(Chains(chain, :parameters).value)
+    pointwise_loglikes(samples, data, ll_fun)
+end
+
+"""
+    pointwise_loglikes(samples::Array{Float64,3}, data, ll_fun)
+
+Computes the pointwise log likelihood 
+
+- `samples`: a three dimensional array of mcmc samples where the first dimension represents the samples, the second dimension represents
+the samples, and the third dimension represents the chains. 
+- `data`: a vector of data used to estimate parameters of the model 
+- `ll_fun`: a function that computes the log likelihood of a single data point: f(θ1, ..., θn, data), where θi is the ith parameter
+"""
+function pointwise_loglikes(samples::Array{Float64,3}, data, ll_fun)
+    fun = (p,d)-> ll_fun(p..., d)
+    n_data = length(data)
+    n_samples, n_chains = size(samples)[[1,3]]
+    pointwise_lls = fill(0.0, n_data, n_samples, n_chains)
+    for c in 1:n_chains 
```

Comment:
This can be done more cleanly with `@tullio`, which will handle this with index notation and will also do the computations much faster than a set of nested `for` loops.

Reply:
I tried to use @tullio but I received a strange error. The code inside the function works, but does not work when called from the function. This is true even when the code does not depend on the arguments. Please see the latest commit for details. I'm probably doing something stupid, but it is not clear to me what that is.